### ISSUE TITLE
Dualstack tests failing

### ISSUE THREAD

**Issue by killianmuldoon** [kubernetes-sigs/cluster-api#9240]
### Which jobs are failing?

https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-e2e-dualstack-and-ipv6-main

### Which tests are failing?

A number of tests are failing consistently.	 Logs etc. are available here: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-cluster-api-e2e-dualstack-and-ipv6-main/1692233880087564288

### Since when has it been failing?

Since August 17th

### Testgrid link

https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-e2e-dualstack-and-ipv6-main

### Anything else we need to know?

The dualstack tests for KIND were failing in the exact same way for a day or so - the solution in CAPI is likely the same as in KIND

Ref: https://testgrid.kubernetes.io/sig-network-kind#sig-network-kind,%20dual,%20master

### Label(s) to be applied

/kind failing-test

@kubernetes-sigs/cluster-api-release-team 
One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels.

**Comment by sbueringer** [comment]
How is this related to #8816?

**Comment by killianmuldoon** [comment]
>How is this related to https://github.com/kubernetes-sigs/cluster-api/issues/8816?

I don't think this is related at all. Different tests failing, and failing consistently.

**Comment by sbueringer** [comment]
@chrischdi If you have some time, would be good if you can take a look. I assume it's a consequence of our 1.28 bump (?)

**Comment by chrischdi** [comment]
Jep, was the only one which was merged in that time window if I got it right. I'll take a look.

**Comment by killianmuldoon** [comment]
@chrischdi  - just to underline this which is probably related:

>The dualstack tests for KIND were failing in the exact same way for a day or so - the solution in CAPI is likely the same as in KIND
Ref: [testgrid.kubernetes.io/sig-network-kind#sig-network-kind,%20dual,%20master](https://testgrid.kubernetes.io/sig-network-kind#sig-network-kind,%20dual,%20master)

**Comment by chrischdi** [comment]
I think this is the related change:

Issue:

- https://github.com/kubernetes/kubernetes/issues/119927

PR:

- https://github.com/kubernetes/kubernetes/pull/119928

**Comment by killianmuldoon** [comment]
So I wonder are we just waiting for a kubekins auto-bump to fix this?

**Comment by chrischdi** [comment]
> So I wonder are we just waiting for a kubekins auto-bump to fix this?

Nope, the conformance image is derived from the kubernetes version.

So we currently use `registry.k8s.io/conformance:v1.28.0`, which does not have the above fix.

**Comment by chrischdi** [comment]
The fix currently get's cherry-picked into release-1.28: 

- https://github.com/kubernetes/kubernetes/pull/119964

**Comment by sbueringer** [comment]
tl;dr if the cherry-pick gets merged, we're good after the 1.28.1 release?

**Comment by chrischdi** [comment]
> tl;dr if the cherry-pick gets merged, we're good after the 1.28.1 release? (and we only have to bump on our side, otherwise no changes needed)

Yes, but with the caveat that: if there will not be a new kindest/node image for 1.28.1: we would always build it in our pipeline.

**Comment by sbueringer** [comment]
I think we could temporarily set the ConformanceImage here: https://github.com/kubernetes-sigs/cluster-api/blob/caaa74482b51fae777334cd7a29595da1c06481e/test/e2e/quick_start_test.go#L124-L128

(and then get rid of it either if we bump to 1.28.{>=1} or 1.29)

**Comment by furkatgofurov7** [comment]
> The fix currently get's cherry-picked into release-1.28:
> 
> * [Automated cherry pick of #119928: e2e WaitForServiceEndpointsNum take into account the address kubernetes/kubernetes#119964](https://github.com/kubernetes/kubernetes/pull/119964)

I reached out to release managers over [slack](https://kubernetes.slack.com/archives/C2C40FMNF/p1692606922601769) asking a help merging it

**Comment by furkatgofurov7** [comment]
> > The fix currently get's cherry-picked into release-1.28:
> > 
> > * [Automated cherry pick of #119928: e2e WaitForServiceEndpointsNum take into account the address kubernetes/kubernetes#119964](https://github.com/kubernetes/kubernetes/pull/119964)
> 
> I reached out to release managers over [slack](https://kubernetes.slack.com/archives/C2C40FMNF/p1692606922601769) asking a help merging it

Looks like, it is a bit late for cherry-pick PR to make it to 1.28.1 due to cherry-picks being in a freeze right now, so it might need to wait for 1.28.2 scheduled in the September patch cycle.

**Comment by sbueringer** [comment]
> Looks like, it is a bit late for cherry-pick PR to make it to 1.28.1 due to cherry-picks being in a freeze right now, so it might need to wait for 1.28.2 scheduled in the September patch cycle.

That's fine. More out of curiosity, are they already release 1.28.1 after the minor release last week? Or are they just releasing patch releases for the other minors?

**Comment by furkatgofurov7** [comment]
> > Looks like, it is a bit late for cherry-pick PR to make it to 1.28.1 due to cherry-picks being in a freeze right now, so it might need to wait for 1.28.2 scheduled in the September patch cycle.
> 
> That's fine. More out of curiosity, are they already releasing 1.28.1 after the minor release last week? Or are they just releasing patch releases for the other minors?

Yes and yes. They cut patch releases from all minor release branches at once on the same day: https://kubernetes.io/releases/patch-releases/#upcoming-monthly-releases
